editor/src/core/shared/element-path.ts

```diff
@@ -642,14 +642,37 @@ export function findAmongAncestorsOfPath<T>(
 }
 
 export function isDescendantOf(target: ElementPath, maybeAncestor: ElementPath): boolean {
-  const targetString = toString(target)
-  const maybeAncestorString = toString(maybeAncestor)
-  return (
-    targetString !== maybeAncestorString &&
-    targetString.startsWith(maybeAncestorString) &&
-    (targetString[maybeAncestorString.length] === SceneSeparator ||
-      targetString[maybeAncestorString.length] === ElementSeparator)
-  )
+  if (target.parts.length >= maybeAncestor.parts.length) {
+    for (let index = 0; index < maybeAncestor.parts.length - 1; index++) {
+      const maybeAncestorPart = maybeAncestor.parts[index]
+      const targetPart = target.parts[index]
+      if (!elementPathPartsEqual(maybeAncestorPart, targetPart)) {
+        return false
+      }
+    }
+    const lastTargetPart = target.parts[maybeAncestor.parts.length - 1]
+    const lastAncestorPart = maybeAncestor.parts[maybeAncestor.parts.length - 1]
+    if (lastTargetPart.length >= lastAncestorPart.length) {
+      for (let partIndex = 0; partIndex < lastAncestorPart.length; partIndex++) {
+        const part = lastAncestorPart[partIndex]
+        if (lastTargetPart[partIndex] !== part) {
+          return false
+        }
+      }
+      if (
+        target.parts.length === maybeAncestor.parts.length &&
+        lastTargetPart.length === lastAncestorPart.length
+      ) {
+        // This caters for the case where the paths are the same.
+        return false
+      }
+      return true
+    } else {
+      return false
+    }
+  } else {
+    return false
+  }
```

Comment:
This is matter of taste, and feel free to ignore, but imho these last two `return false` else branches would be much more readable at the beginning, according to the "happy path to the left" principle (right now you need to search back and find the actual indentation level to check what does this else mean)

Reply:
For whatever reason I find it easier (within reason) to have it more nested. I've found especially when I'm changing the code a lot that early returns can be a roadblock as I have to rework them into regular if..else statements.

Reply:
I have to agree with Balint, I opened this resolved comment because I was about to comment here myself, that at least here in the diff viewer this does not look refactor-proof to me: 
<img width="250" alt="image" src="https://github.com/user-attachments/assets/97ffcc1f-139a-41ff-af04-590f3073133d">

Reply:
Yeah I'm going to unresolve this comment, because I came here for the same reason. Historically I've been against early returns, but I certainly prefer the "happy path to the left" principle that we've been following in the code for the past year+. I think it's fair to say that that preference is now shared by the majority of the team (based on conversations with Berci and Federico too), so please can we collectively stick to it.

Reply:
Thank you (and sorry!)